manifests/config.pp

```diff
@@ -59,6 +60,43 @@
 
     }
 
+    'Suse': {
+      $pamconfig_mkhomedir_check_cmd = '/usr/sbin/pam-config -q --mkhomedir | grep session:'
+      $pamconfig_check_cmd           = '/usr/sbin/pam-config -q --sss | grep session:'
+
+      if $mkhomedir {
+
+        exec { 'pam-config -a --mkhomedir':
+          path   => '/bin:/usr/bin:/sbin:/usr/sbin',
+          unless => $pamconfig_mkhomedir_check_cmd,
+        }
+
+      }
+
+      exec { 'pam-config -a --sss':
+        path   => '/bin:/usr/bin:/sbin:/usr/sbin',
+        unless => $pamconfig_check_cmd,
+      }
+
+      if $manage_nsswitch {
+
```

Comment:
I just tested the module trlinkin-nsswitch. The problem is that this module will set default values and sss for earch entry.
Also according to this module the osfamily operatingsystem SLES does not have an entry sudoers or shadow. That's why a concat on this default value and sss is failing.